docs/source/llms/tracing/notebooks/jupyter-trace-demo.ipynb

```diff
@@ -0,0 +1,379 @@
+   "id": "36c37b34",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Simple manual tracing example\n",
+    "import mlflow\n",
+    "\n",
+    "\n",
+    "@mlflow.trace\n",
+    "def foo(input):\n",
+    "    return input + 1\n",
+    "\n",
+    "\n",
+    "# running foo() generates a trace\n",
+    "foo(1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "76ab09c7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Automatic tracing with OpenAI\n",
+    "import os\n",
+    "from getpass import getpass\n",
+    "\n",
+    "os.environ[\"OPENAI_API_KEY\"] = getpass(\"Enter your OpenAI API key: \")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "053b397f-b816-4bb0-bb9c-103124f65385",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from openai import OpenAI\n",
+    "\n",
+    "import mlflow\n",
+    "\n",
+    "mlflow.openai.autolog()\n",
+    "\n",
+    "client = OpenAI()\n",
+    "\n",
+    "# creating a chat completion will generate a trace\n",
+    "client.chat.completions.create(\n",
+    "    model=\"gpt-4o-mini\",\n",
+    "    messages=[{\"role\": \"user\", \"content\": \"hello!\"}],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ec2ddd6c",
+   "metadata": {},
+   "source": [
+    "### Example 2: Displaying a Trace object\n",
+    "\n",
+    "The trace UI will also show up when an MLflow Trace entity is displayed. This can happen in two ways:\n",
+    "\n",
+    "1. Explicitly displaying a trace object with IPython's `display()`\n",
+    "2. When a trace object happens to be the last evaluated expression in a cell"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "24faa8e3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Explicitly calling `display()`\n",
+    "trace = mlflow.get_last_active_trace()\n",
+    "display(trace)\n",
+    "\n",
+    "# Even if the last expression does not result in a trace,\n",
+    "# display(trace) will still trigger the UI display\n",
+    "print(\"Test\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7fb27b941602401d91542211134fc71a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Displaying as a result of the trace being the last expression\n",
+    "trace"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "782b3f4b",
+   "metadata": {},
+   "source": [
+    "### Example 3: Calling `mlflow.search_traces()`\n",
+    "\n",
+    "MLflow provides the `mlflow.search_traces()` API to conveniently search through all traces in an experiment. When this API is called in a Jupyter notebook, the trace UI will render all the traces in a paginated view. There is a limit to how many traces can be rendered in a single cell output--by default, the maximum is 10. However, this can be configured by setting the `MLFLOW_MAX_TRACES_TO_DISPLAY_IN_NOTEBOOK` environment variable."
```

Comment:
```suggestion
    "MLflow provides the `mlflow.search_traces()` API to conveniently search through all traces in an experiment. When this API is called in a Jupyter notebook, the trace UI will render all the traces in a paginated view. There is a limit to how many traces can be rendered in a single cell output by default, the maximum is 10. However, this can be configured by setting the `MLFLOW_MAX_TRACES_TO_DISPLAY_IN_NOTEBOOK` environment variable."
```

Is this a typo?

Reply:
it's my writing style 😂 i like to use em-dashes but i can remove